File: 0881-boats-to-save-people/0881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        peoples=sorted(people)
        left,right,count=0,len(peoples)-1,0
        while left<=right:
            if peoples[left]+peoples[right]<=limit:
                left+=1
                count+=1
            else:
                count+=1
            right-=1
        return count
        

File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        int l=0;
        int n=people.size();
        int r=n-1;
        int ans=0;
        sort(people.begin(),people.end());
        while(l<r){
            if(people[l]+people[r]<=limit){
                people[l]=-1;
                people[r]=-1;
                l++;
                r--;
                ans++;
            }
            else if(people[l]+people[r]>limit){
                r--;
            }
            else l++;
        }
        for(auto it:people){
            if(it!=-1){
                ans++;
            }
        }
        return ans;
    }
};